### PR TITLE
avoid multiple registrations of the same method

### LINKED ISSUE #657: same method can be registered multiple times
The plugin registry does not check if the method is already registered before registering it again.

## Expected Behavior
It should silently ignore the registration of a previously registered method.

## Current Behavior
It adds the method.

### PR DESCRIPTION
**Pull request by jpaniagualaconich**
Fixes #657 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by claudep** [comment]
Once again, it would be great to add a small regression test.

**Comment by jpaniagualaconich** [comment]
@claudep added `tests/test_registry.py`

File: two_factor/plugins/registry.py [hunk 1]
```diff
@@ -77,6 +77,10 @@ def __init__(self):
         self.register(GeneratorMethod())
 
     def register(self, method):
+        for registered_method in self._methods:
+            if method.code == registered_method.code:
+                return   # Already registered, ignore.
```
Comment (claudep) [inline two_factor/plugins/registry.py:82]: Are you OK with the comment I added?

Comment (jpaniagualaconich) [reply two_factor/plugins/registry.py:82]: totally :+1:

**Review by claudep — APPROVED** [review]

**Pull request merged**